doc/sphinx/language/gallina-extensions.rst

```diff
@@ -869,167 +866,137 @@ Module system
-   :name: Inline
+.. todo: description below applies equally to modules and module types?
 
-   The instance of this assumption will be automatically expanded at functor application, except when
-   this functor application is prefixed by a ``!`` annotation.
+Interactive modules are terminated with the :cmd:`End` command, which
+is also used to terminate :ref:`Sections<section-mechanism>`.
+:n:`End @ident` closes the interactive module or module type :token:`ident`.
+If the module type was given, the command verifies that the content of the module
+matches the module type.  If the module is basic (not a
+functor), its components (constants, inductive types, submodules etc.)
+are now available through the dot notation.
 
-.. cmd:: End @ident
+.. exn:: No such label @ident.
+    :undocumented:
 
-   This command closes the interactive module type :token:`ident`.
+.. exn:: Signature components for label @ident do not match.
+    :undocumented:
 
-   .. exn:: This is not the last opened module type.
-      :undocumented:
+.. exn:: This is not the last opened module.
+    :undocumented:
 
-.. cmd:: Module Type @ident := @module_type
+.. exn:: This is not the last opened module type.
+   :undocumented:
 
-   Defines a module type :token:`ident` equal to :token:`module_type`.
+.. cmd:: Include @module_type_inl {* <+ @module_expr_inl }
 
-    .. cmdv:: Module Type @ident {* @module_binding} := @module_type
+   Includes the content of module(s) in the current
+   interactive module. Here :n:`@module_type_inl` can be a module expression or a module
+   type expression. If it is a high-order module or module type
+   expression then the system tries to instantiate :n:`@module_type_inl` with the current
+   interactive module.
 
-       Defines a functor type :token:`ident` specifying functors taking arguments :token:`module_bindings` and
-       returning :token:`module_type`.
+   Including multiple modules is a single :cmd:`Include` is equivalent to including each module
+   in a separate :cmd:`Include` command.
 
-    .. cmdv:: Module Type @ident {* @module_binding} := {+<+ @module_type }
+.. cmd:: Include Type @module_type_inl {* <+ @module_type_inl }
 
-       is equivalent to an interactive module type were each :token:`module_type` is included.
+   .. deprecated:: Use :cmd:`Include` instead.
 
-.. cmd:: Declare Module @ident : @module_type
+.. cmd:: Declare Module {? {| Import | Export } } @ident {* @module_binder } : @module_type_inl
 
-   Declares a module :token:`ident` of type :token:`module_type`.
+   Declares a module :token:`ident` of type :token:`module_type_inl`.
 
-    .. cmdv:: Declare Module @ident {* @module_binding} : @module_type
+   If :n:`@module_binder`\s are specified, declares a functor with parameters given by the list of
+   :token:`module_binder`\s.
 
-       Declares a functor with parameters given by the list of :token:`module_binding` and output module type
-       :token:`module_type`.
+.. _example_def_simple_module:
```

Comment:
```suggestion
Examples
~~~~~~~~

.. _example_def_simple_module:
```

Same thing as above. We need to show that we are out of the previous sub-section.

Reply:
I think the ambiguity of where subsections end is a general problem we should give some thought to.  We could eventually have a dividing line or center something like "* * *" for use where it helps.

How about if I move the example to the end of the module section?  There's already a hyperlink to the example in `Module`.

Reply:
Yes to moving examples at the end. I think that's what I suggested somewhere.